system-addon/lib/TopStoriesFeed.jsm

```diff
@@ -28,8 +28,10 @@ this.TopStoriesFeed = class TopStoriesFeed {
       const prefs = new Prefs();
       const options = JSON.parse(prefs.get("feeds.section.topstories.options"));
       const apiKey = this._getApiKeyFromPref(options.api_key_pref);
-      this.stories_endpoint = this._produceUrlWithApiKey(options.stories_endpoint, apiKey);
-      this.topics_endpoint = this._produceUrlWithApiKey(options.topics_endpoint, apiKey);
+      const locale = Services.locale.getRequestedLocale();
+      this.stories_endpoint = this._produceFinalEndpointUrl(options.stories_endpoint, apiKey, locale);
```

Comment:
I see that `fetch*` is called later in `init`, but does this actually need to happen during init? We ran into an issue where reftests were failing because network connections were being made aggressively (and denied).

But even ignoring the network connection issue (already fixed for now), instead of using `Services.locale.getRequestedLocale`, `onAction` should watch for `LOCALE_UPDATED` and have a helper method set the endpoint and maybe fetch. The helper would be called from LOCALE_UPDATED and `init` and can get the locale from `this.store.getState().App.locale`